### PR TITLE
port system.diagnostics.performanceCounter

### PR DESCRIPTION
**Pull request by Anipik**
- set privateAssets = all to avoid adding package dependencies on ns2.0
- add _GetPackageFiles to before targets for NETStandardCompatError

Another example pr for reference.

https://github.com/Anipik/PackagePortTester  

i pushed the package validator which could be run on package paths to test if the port is successful.

### CODE REVIEW

File: src/libraries/System.Diagnostics.PerformanceCounter/src/System.Diagnostics.PerformanceCounter.csproj [hunk 1]
```diff
@@ -150,8 +151,12 @@
     <Reference Include="System.Threading.Thread" />
   </ItemGroup>
   <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
-    <PackageReference Include="System.Buffers" Version="$(SystemBuffersVersion)" />
-    <PackageReference Include="System.Memory" Version="$(SystemMemoryVersion)" />
-    <ProjectReference Include="$(LibrariesProjectRoot)System.Runtime.CompilerServices.Unsafe\src\System.Runtime.CompilerServices.Unsafe.ilproj" />
+    <PackageReference Include="System.Buffers" Version="$(SystemBuffersVersion)" PrivateAssets="all" />
+    <PackageReference Include="System.Memory" Version="$(SystemMemoryVersion)" PrivateAssets="all" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Runtime.CompilerServices.Unsafe\src\System.Runtime.CompilerServices.Unsafe.ilproj" PrivateAssets="all" />
```
Comment (ViktorHofer) [inline src/libraries/System.Diagnostics.PerformanceCounter/src/System.Diagnostics.PerformanceCounter.csproj:156]: Why are and were these dependencies omitted from the package? i.e. the 5.0.1 version doesn't have them either: https://www.nuget.org/packages/System.Diagnostics.PerformanceCounter/5.0.1.

File: Directory.Build.targets [hunk 2]
```diff
@@ -77,7 +77,7 @@
           Condition="'@(NETStandardCompatError)' != ''"
           Inputs="%(NETStandardCompatError.Identity)"
           Outputs="unused"
-          BeforeTargets="GetFiles">
+          BeforeTargets="_GetPackageFiles;GetFiles">
```
Comment (ViktorHofer) [inline Directory.Build.targets:80]: Why is this change necessary? Wasn't this working on csproj before?

Comment (Anipik) [reply Directory.Build.targets:80]: No it wasn't working on the csproj, the extensions projects like Microsoft.Extensions.Primitives were missing this file.
GetFiles is a custom target defined by our old pkgproj system and is not used by dotnet pack.

File: src/libraries/System.Diagnostics.PerformanceCounter/src/System.Diagnostics.PerformanceCounter.csproj (hunk 1 above)
Comment (Anipik) [reply src/libraries/System.Diagnostics.PerformanceCounter/src/System.Diagnostics.PerformanceCounter.csproj:156]: Yes these dependencies are omitted from the package. None of previous versions have these dependencies.  I went through the binlog and was not able to find why the old packaging system is removing this.

File: Directory.Build.targets (hunk 2 above)
Comment (ViktorHofer) [reply Directory.Build.targets:80]: I see. Can we avoid the custom target here and instead just use `Pack`?

File: src/libraries/System.Diagnostics.PerformanceCounter/src/System.Diagnostics.PerformanceCounter.csproj (hunk 1 above)
Comment (ViktorHofer) [reply src/libraries/System.Diagnostics.PerformanceCounter/src/System.Diagnostics.PerformanceCounter.csproj:156]: I don't see references to these assemblies in its metadata either:

```
// Detected Target-Framework-Id: .NETStandard,Version=v2.0

// Referenced assemblies (in metadata order):
// netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51 (unresolved)
```

Can you please check why those references are necessary for compilation?

Comment (Anipik) [reply src/libraries/System.Diagnostics.PerformanceCounter/src/System.Diagnostics.PerformanceCounter.csproj:156]: Nope they are not required for compilation

**Review by ViktorHofer — APPROVED** [review]

**Comment by ViktorHofer** [comment]
Failures are due to work items being canceled which I pinged dnceng about in another issue.
